components/hab/src/command/cli/setup.rs

```diff
@@ -311,3 +353,66 @@ fn opt_out_analytics(ui: &mut UI, analytics_path: &Path) -> Result<()> {
     ui.br()?;
     result
 }
+
+#[cfg(windows)]
+fn ask_set_binlink_path(ui: &mut UI) -> Result<bool> {
+    Ok(ui.prompt_yes_no("Add binlink directory to PATH?", Some(true))?)
+}
+
+#[cfg(not(target_os = "windows"))]
+fn ask_set_binlink_path(ui: &mut UI) -> Result<bool> {
+    Ok(false)
+}
+
+#[cfg(windows)]
+fn binlink_is_on_path(binlink_path: &Path) -> bool {
+    let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+    if let Ok(env) = hklm.open_subkey_with_flags(
+        "System\\CurrentControlSet\\Control\\Session Manager\\Environment",
+        KEY_READ,
+    ) {
+        let path: String = env
+            .get_value("path")
+            .expect("could not find a machine PATH");
+        return env::split_paths(&path).any(|p| p == binlink_path);
+    }
+    false
+}
+
+#[cfg(windows)]
+fn set_binlink_path(binlink_path: &Path) -> Result<()> {
+    if !binlink_is_on_path(binlink_path) {
+        // this sets the permanent machine PATH and not
+        // the path of this process. These are maintained
+        // in the Windows registry
+        let hklm = RegKey::predef(HKEY_LOCAL_MACHINE);
+        let env = hklm.open_subkey_with_flags(
+            "System\\CurrentControlSet\\Control\\Session Manager\\Environment",
+            KEY_ALL_ACCESS,
+        )?;
+        let path: String = env.get_value("path")?;
+        let new_path = format!("{};{}", binlink_path.display(), path);
+        env.set_value("path", &new_path)?;
+
+        // After altering te machine environment, we must broadcast
+        // a WM_SETTINGCHANGE message to all windows so the user
+        // will not need to sign out/in for the new path to take effect
+        unsafe {
+            user32::SendMessageTimeoutW(
+                winapi::HWND_BROADCAST,
+                winapi::WM_SETTINGCHANGE,
+                0,
+                WideCString::from_str("Environment").unwrap().as_ptr() as winapi::LPARAM,
+                winapi::SMTO_ABORTIFHUNG,
+                5000,
+                ptr::null_mut(),
+            );
+        }
+    }
+    Ok(())
+}
+
+#[cfg(not(target_os = "windows"))]
+fn set_binlink_path(binlink_path: &Path) -> Result<()> {
+    Ok(())
```

Comment:
I think [`unreachable!()`](http://doc.rust-lang.org/1.28.0/core/macro.unreachable.html) may be more appropriate here, since I think this would be a logic error if we called this from a non-windows context.